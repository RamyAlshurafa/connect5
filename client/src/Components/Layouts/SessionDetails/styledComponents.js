import styled from "styled-components";
import { NavLink } from "react-router-dom";

const Wrapper = styled.div`
  background: #eff3f4;
  display: Grid;
  grid-template-columns: 1fr;
  grid-template-areas:
    "header"
    "content"
    "actions";
  margin-bottom: 72px;
`;

const Header = styled.div`
  grid-area: header;
  position: relative;
  color: #0b6fa4;
  font-size: 15px;
  @media (min-width: 1040px) {
    width: 80%;
    margin: 0 auto;
  }
`;

const Heading = styled.div`
  color: #0b6fa4;
  width: 100%;
  padding-left: 10px;
`;

const Borderbottom = styled.div`
  width: 230px;
  height: 1px;
  position: absolute;
  background: #000;
  top: 60px;
  background: #0b6fa4;
  margin-left: 10px;
`;

const Content = styled.div`
  grid-area: content;
  display: flex;
  flex-direction: column;
  background: #fff;
  border-radius: 10px;
  top: 100px;
  margin: 15px 10px 0 10px;
  @media (min-width: 1040px) {
    width: 80%;
    margin: 0 auto;
  }
`;

const Statistic = styled.div`
  display: flex;
  flex-direction: row;
  width: 80%;
  height: 25.3%;
  margin: 0 auto;
  margin-top: 15px;
  border-bottom: 1px solid #a8a193
  > :last-child {
    border-right: none;
  }
`;

const StatisticItems = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 85%;
  border-right: 1px solid #a8a193;
  padding-right: 5px;
`;

const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 90%;
  height: 50%;
  background: #c9dae18a;
  border-radius: 5px;
  margin: 0 auto;
  margin-top: 10px;
  padding-bottom: 10px;
`;

const LinkType = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 25%;
  padding: 15px 0 0;
  justify-content: space-between;
`;
const ItemName = styled.div`
  display: flex;
  margin: 0 auto;
  height: 65%;
  margin-top: 7px;
  font-size: 15px;
  color: #1397b1;
  padding-bottom: 10px;
`;

const Items = styled.div`
  display: flex;
  margin: 0 auto;
  height: 35%;
  font-size: 15px;
  color: #1397b1;
  padding-bottom: 10px;
`;

const Links = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  @media (min-width: 1040px) {
    width: 80%;
    margin: 0 auto;
  }
`;
const SurveyLink1 = styled(Links)`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
`;

const LinkInfo = styled.span`
  font-size: 15px;
  margin-left: 5px;
`;

const SurveyType = styled.span`
  padding: 0 5px;
`;

const Span1 = styled.div`
  font-size: 18px;
  display: flex;
  cursor: pointer;
  padding-left: 6px;
`;

const Span2 = styled.div`
  font-size: 18px;
  display: flex;
  cursor: pointer;
`;

const copyLink = styled.div`
  font-size: 15px;
  margin-left: 5px;
  margin-top: -7px;
`;

const SurveyLink = styled.div`
display: flex;
 width: 100%;
 height: 100%;
 margin-top: 0;
 margin-bottom: 20px;
`;

const SLink = styled.a`
  text-decoration: none;
  display: block;
  padding: 10px 10px 10px 25px;
  margin-left: 15px;
  color: #3d3d3df2;
  cursor: default;
  word-break: break-all;

  & :hover {
    color: blue;
    text-decoration: underline;
  }
`;

const Button = styled.button`
  color: white;
  margin: 0 auto;
  margin-top: 15px;
  height: 50px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 14px;
  cursor: pointer;
  border-radius: 5px;
  & :hover {
    background-color: white;
    border: 1px solid #2a617b;
    color: #2a617b;
  }
`;

const ActionsWrapper = styled.div`
  grid-area: actions;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin: 12px auto;
  
`;

const ActionWrapper = styled.div`
  display: flex;
`;

const ActionBtn = styled.button`
  background: none;
  border: none;
  width: 40px;
`;

const Icon = styled.i`
  font-size: 18px;
`;

const Span = styled.span`
  line-height: 25px;
`;

const Btn = styled.button`
  background: none;
  border: none;
  display: flex;
`;

const ResultsBtn = styled.button`
  background: #42c4de;
  border: 1px solid #42c4de;
  margin: 0 auto;
  padding: 10px 3px;
  width: 140px;
  display: inline-block;
  color: white;
  text-decoration: none;
  -webkit-border-radius: 2.5em;
  -moz-border-radius: 2.5em;
  border-radius: 2em;
  margin-bottom: 6px;
`;

const Navlink = styled(NavLink)`
  display: block;
  text-decoration: none;
  padding: 0 12px;
  color: white;
`;
const IconsContainer = styled.div`

display: flex;
`;

export {
  Navlink,
  Span,
  Span1,
  Span2,
  Icon,
  ActionBtn,
  ActionWrapper,
  ActionsWrapper,
  Button,
  ResultsBtn,
  SLink,
  SurveyLink,
  LinkType,
  LinkInfo,
  Links,
  Items,
  ItemName,
  StatisticItems,
  Statistic,
  Content,
  Borderbottom,
  Header,
  Heading,
  Wrapper,
  SurveyLink1,
  Btn,
  copyLink,
  Container,
  SurveyType,
  IconsContainer,
};
