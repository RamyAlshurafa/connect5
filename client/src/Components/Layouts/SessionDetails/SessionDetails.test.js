import React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router-dom";

import SessionDetails from "./index";

test("SessionDetails matches snapshot", () => {
  const context = {};
  const object = {
    date: 123,
    type: 1,
    attendees: 10,
    surveyURL1: "123",
    surveyURL2: "123",
  };
  const component = renderer.create(
    <StaticRouter context={context}>
      <SessionDetails sessionDetails={object} />
    </StaticRouter>,
  );
  console.log(component);
  expect(component).toMatchSnapshot();
});
