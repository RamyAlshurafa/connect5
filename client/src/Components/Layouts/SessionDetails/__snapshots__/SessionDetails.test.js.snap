// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionDetails matches snapshot 1`] = `
<div
  className="sc-bdVaJa jExVmn"
>
  <div
    className="sc-bwzfXH ekyToE"
  >
    <div
      className="sc-htpNat dDDduC"
    >
      <h1>
        Session Details
      </h1>
    </div>
    <div
      className="sc-bxivhb cMkSyr"
    />
  </div>
  <div
    className="sc-ifAKCX ivJThE"
  >
    <div
      className="sc-EHOje etMTDo"
    >
      <div
        className="sc-bZQynM fTLMmD"
      >
        <div
          className="sc-dnqmqq jGEbHI"
        >
          Date
        </div>
        <div
          className="sc-iwsKbI hQYCHX"
        >
          01/01/1970
        </div>
      </div>
      <div
        className="sc-bZQynM fTLMmD"
      >
        <div
          className="sc-dnqmqq jGEbHI"
        >
          Type
        </div>
        <div
          className="sc-iwsKbI hQYCHX"
        >
          1
        </div>
      </div>
      <div
        className="sc-bZQynM fTLMmD"
      >
        <div
          className="sc-dnqmqq jGEbHI"
        >
          Attendees
        </div>
        <div
          className="sc-iwsKbI hQYCHX"
        >
          10
        </div>
      </div>
    </div>
    <div
      className="sc-gZMcBi hRbZzM"
    >
      <div
        className="sc-kAzzGY brdSJE"
      >
        <div
          className="sc-gZMcBi sc-gqjmRU ecWkXe"
        >
          <div
            className="sc-gzVnrw hFxObc"
          >
            <div
              className="sc-htoDjs cTWPSR"
            >
              <div
                className="sc-fjdhpX ejPixh"
              >
                <span
                  className="sc-jTzLTM gkQkbK"
                >
                  Pre-Survey
                </span>
                Link
              </div>
              <div
                className="sc-kkGfuU fbneOr"
              >
                <div
                  className="sc-jzJRlG bEvwBw"
                  onClick={[Function]}
                >
                  <i
                    className="fas fa-info-circle"
                  />
                  <span
                    className="sc-VigVT gGKHMo"
                  >
                    info
                  </span>
                </div>
                <copyLink>
                  <button
                    className="sc-eNQAEJ eEQhZo"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <i
                      className="far fa-clone"
                    />
                    <span
                      className="sc-VigVT gGKHMo"
                    >
                      copy
                    </span>
                  </button>
                </copyLink>
              </div>
            </div>
            <a
              className="sc-chPdSV lfKpfA"
              href="localhost/survey/0undefined"
            >
              localhost/survey/0undefined
            </a>
            <button
              className="sc-hMqMXs gAuTYf"
            >
              <a
                aria-current={null}
                className="sc-kEYyzF bLOOzH"
                href="/session/details/undefined/1"
                onClick={[Function]}
              >
                 Survey Results 
              </a>
            </button>
          </div>
          <div
            className="sc-gzVnrw hFxObc"
          >
            <div
              className="sc-htoDjs cTWPSR"
            >
              <div
                className="sc-fjdhpX ejPixh"
              >
                <span
                  className="sc-jTzLTM gkQkbK"
                >
                  Survey 1
                </span>
                Link
              </div>
              <div
                className="sc-kkGfuU fbneOr"
              >
                <div
                  className="sc-jzJRlG bEvwBw"
                  onClick={[Function]}
                >
                  <i
                    className="fas fa-info-circle"
                  />
                  <span
                    className="sc-VigVT gGKHMo"
                  >
                    info
                  </span>
                </div>
                <copyLink>
                  <button
                    className="sc-eNQAEJ eEQhZo"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <i
                      className="far fa-clone"
                    />
                    <span
                      className="sc-VigVT gGKHMo"
                    >
                      copy
                    </span>
                  </button>
                </copyLink>
              </div>
            </div>
            <a
              className="sc-chPdSV lfKpfA"
              href="localhost/survey/1undefined"
            >
              localhost/survey/1undefined
            </a>
            <button
              className="sc-hMqMXs gAuTYf"
            >
              <a
                aria-current={null}
                className="sc-kEYyzF bLOOzH"
                href="/session/details/undefined/1"
                onClick={[Function]}
              >
                 Survey Results 
              </a>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-kGXeez hmiuWc"
  >
    <div
      className="sc-kpOJdX ieBFuD"
    >
      <button
        className="sc-dxgOiQ hvurjS"
        onClick={[Function]}
        type="button"
      >
        <i
          className="far fa-edit sc-ckVGcZ ewqYsR"
        />
      </button>
      <span
        className="sc-jKJlTe libXlj"
      >
        edit session
      </span>
    </div>
    <div
      className="sc-kpOJdX ieBFuD"
    >
      <button
        className="sc-dxgOiQ hvurjS"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-trash-alt sc-ckVGcZ ewqYsR"
        />
      </button>
      <span
        className="sc-jKJlTe libXlj"
      >
        delete session
      </span>
    </div>
  </div>
</div>
`;
